### PR TITLE
[SYCL] Add -foffload-fp32-prec-[div/sqrt] FE option handling in JIT

### PR DESCRIPTION
**Pull request by MrSidims**
It matches -cl-fp32-correctly-rounded-divide-[div/sqrt] option for OpenCL and -ze-fp32-correctly-rounded-divide-[div/sqrt] for L0.

The option itself is being added in https://github.com/intel/llvm/pull/15836 .

### CODE REVIEW

File: source/adapters/cuda/platform.cpp [hunk 1]
```diff
@@ -168,5 +168,10 @@ UR_APIEXPORT ur_result_t UR_APICALL urPlatformGetBackendOption(
     *ppPlatformOption = "";
     return UR_RESULT_SUCCESS;
   }
+  if (pFrontendOption == "-foffload-fp32-prec-div" ||
+      pFrontendOption == "-foffload-fp32-prec-sqrt") {
+    *ppPlatformOption = "";
```
Comment (MrSidims) [inline source/adapters/cuda/platform.cpp:173]: TBH I'm not sure if it's needed.

**Review by aarongreig — APPROVED** [review]

File: source/adapters/cuda/platform.cpp (hunk 1 above)
Comment (jchlanda) [reply source/adapters/cuda/platform.cpp:173]: It's fine for JIT, but you are right should we want to support it there, it would require more work.

Comment (MrSidims) [reply source/adapters/cuda/platform.cpp:173]: So it's fine to leave it as is?

**Comment by MrSidims** [comment]
@npmiller @aarongreig @jchlanda @omarahmed1111 please approve workflows to run

File: source/adapters/cuda/platform.cpp (hunk 1 above)
Comment (jchlanda) [reply source/adapters/cuda/platform.cpp:173]: Yeap, I would say so.

**Review by jchlanda — APPROVED** [review]
CUDA/HIP 👍

**Comment by MrSidims** [comment]
@npmiller @aarongreig @jchlanda @omarahmed1111 please approve one more type workflows to run

**Review by jchlanda — APPROVED** [review]

**Comment by MrSidims** [comment]
Apologies @npmiller @aarongreig @jchlanda @omarahmed1111 I have to ask to approve one more time workflows to run, hope this time it should be fine

**Comment by MrSidims** [comment]
For level_zero pre-commit:
Failed Tests (2):
  SYCL :: Matrix/joint_matrix_bf16_fill_k_cache_arg_dim.cpp
  SYCL :: Matrix/joint_matrix_bf16_fill_k_cache_runtime_dim.cpp

********************
Unexpectedly Passed Tests (3):
  SYCL :: ESIMD/mask_expand_load.cpp
  SYCL :: Matrix/SPVCooperativeMatrix/SG32/joint_matrix_apply_two_matrices.cpp
  SYCL :: Matrix/SPVCooperativeMatrix/joint_matrix_apply_two_matrices.cpp
  
  Is unrelated to this change.
  
OpenCL E2E results: I've seen exactly the same failures in https://github.com/oneapi-src/unified-runtime/actions/runs/11835346462/job/32979251942?pr=2326 , so it's also unrelated.
  
CUDA E2E results are also unlikely to relate with the patch, but here I'm not an expert to say if there is an unexpected interaction. I've seen that in other PRs the results of the runs on CUDA might be inconsistent.

  @npmiller @aarongreig @jchlanda @omarahmed1111  based on this, can we merge the PR?

**Comment by aarongreig** [comment]
looks like it still needs an approval from @oneapi-src/unified-runtime-level-zero-write, then we can add the ready-to-merge label to get it in the queue

**Comment by MrSidims** [comment]
@aarongreig got it, thanks. Whom can I ask to take a look from that team perspective? Note, as I'm not listed in repo contributors I can't add new reviewers. Also when I start typing @oneapi-src.... - github doesn't find any account with such name.

**Comment by MrSidims** [comment]
@pbalcer @nrspruit could you please take a look from level zero rt perspective?

**Review by nrspruit — APPROVED** [review]
LGTM, I needed to lookup the flags to ensure that was the correct ones for L0 in IGC. It all looks fine to me, thanks!

**Comment by MrSidims** [comment]
@nrspruit thanks! What should be done next to make the PR be merged? Note, I can not add the mentioned by @aarongreig label to the PR.

**Comment by MrSidims** [comment]
@npmiller @aarongreig @jchlanda @omarahmed1111 @nrspruit what should I do to make the patch merged?

**Comment by MrSidims** [comment]
Hi @callumfare @npmiller @aarongreig @jchlanda @omarahmed1111 @nrspruit what should I do to make the patch merged?

**Comment by callumfare** [comment]
Hi @MrSidims, it looks like this PR should be in by the end of the day.  We have a merge queue system where we merge PRs in the order they are marked ready to merge, updating the UR tag in intel/llvm each time where needed. This keeps the changes between the two repos in sync but can take a little while to get through everything. If a PR is high priority we can bump it to the top of the queue.

**Comment by MrSidims** [comment]
@callumfare thanks for the explanation. I've just seen several more fresh PRs merged ahead of this PR and started wondering if I'm missing something.

**Pull request merged**